### PR TITLE
Added tcp transport user documentation.

### PR DESCRIPTION
**Pull request by LuisGP**
Added TCP to lists of transports available.
Added TCP configuration example.
Added custom transport definition and usage with XML.
Changed locator.set_IP4_address methods to the new IPLocator::setIPv4 methods.

### CODE REVIEW

File: docs/pubsub.rst [hunk 1]
```diff
@@ -468,12 +468,98 @@ Also you can disable built-in UDPv4 network transport using the attribute ``rtps
...
+participant.
+
+You can learn more about profiles in xml in :ref:`xml-profiles`.
+
+TCP Transport Configuration
+"""""""""""""""""""""""""""
+
+To use TCP as transport you need to define some more configurations:
+
+First, you need to indicate Fast-RTPS to not use default transports (if you only want to use TCP) through
```
Comment (JuanCarlos-eProsima) [inline docs/pubsub.rst]: "if you only want to use TCP" can create misunderstandings because this flag can be used with other transports or configurations

Comment (LuisGP) [reply docs/pubsub.rst]: You are right. I think the best option is to remove the full sentence, as *useBuiltinTransports* is explained already.

File: docs/advanced.rst [hunk 2]
```diff
@@ -144,11 +144,15 @@ by providing an alternative configuration when you create the Participant.
...
     Pparams.userTransports.push_back(my_transport); //Link the Transport Layer to the Participant
 
+See :ref:`network-configuration` for TCP examples.
+
 Note that unless you manually disable the built-in transport layer, the Participant will use
 your custom transport configuration along the built-in one.
 
 This distribution comes with an example of how to change the configuration of the transport layer. It can be found `here <https://github.com/eProsima/Fast-RTPS/tree/master/examples/C%2B%2B/UserDefinedTransportExample>`_.
 
+Also a TCP version of helloworld example can be found in this `link <https://github.com/eProsima/Fast-RTPS/tree/master/examples/C%2B%2B/HelloWorldExampleTCP>`_.
```
Comment (richiware) [inline docs/advanced.rst]: Please, join this paragraph with the previous.

File: docs/pubsub.rst [hunk 3]
```diff
@@ -468,12 +468,95 @@ Also you can disable built-in UDPv4 network transport using the attribute ``rtps
...
+                ...
+            </rtps>
+        </participant>
+        ...
+    </profiles>
+
+To add a custom transport to a participant, you must indicate its transport_id inside the userTransports labels of the
+participant.
+
+You can learn more about profiles in xml in :ref:`xml-profiles`.
```
Comment (richiware) [inline docs/pubsub.rst]: Join this paragraph with the previous.

File: docs/advanced.rst [hunk 4]
```diff
@@ -134,7 +134,7 @@ Transport Layer
 ---------------
```
Comment (richiware) [inline docs/advanced.rst]: Remove this section about transports. This information is already in advance configuration of pubsub.rst.

File: docs/pubsub.rst [hunk 5]
```diff
@@ -468,12 +468,95 @@ Also you can disable built-in UDPv4 network transport using the attribute ``rtps
...
-Network endpoints are defined by *eProsima Fast RTPS* as locators. Locators in *eProsima Fast RTPS* are enclosed as type :class:`Locator_t`, which has the following fields:
+Network endpoints are defined by *eProsima Fast RTPS* as locators.
+Locators in *eProsima Fast RTPS* are enclosed as type :class:`Locator_t`, which has the following fields:
 
-* ``kind``: Defines the protocol. *eProsima Fast RTPS* currently supports UDPv4 or UDPv6
+* ``kind``: Defines the protocol. *eProsima Fast RTPS* currently supports UDPv4, UDPv6, TCPv4 or TCPv6
 * ``port``: Port as an UDP/IP port.
 * ``address``: Maps to IP address
 
+If you are configuring your application using a XML file, you can add custom transports too, inside the profiles
```
Comment (richiware) [inline docs/pubsub.rst]: Move this to TCP transport configuration.

File: docs/pubsub.rst [hunk 6]
```diff
@@ -468,12 +468,95 @@ Also you can disable built-in UDPv4 network transport using the attribute ``rtps
...
+        </participant>
+        ...
+    </profiles>
+
+To add a custom transport to a participant, you must indicate its transport_id inside the userTransports labels of the
+participant.
+
+You can learn more about profiles in xml in :ref:`xml-profiles`.
+
+TCP Transport Configuration
```
Comment (richiware) [inline docs/pubsub.rst]: Create a parent section about transports.

File: docs/pubsub.rst [hunk 7]
```diff
@@ -440,14 +440,14 @@ Advanced configuration
...
-transports: UDPv4 and UDPv6. By default, when a :class:`Participant` is created, one built-in UDPv4 network transport is
-configured.
+*eProsima Fast RTPS* implements an architecture of pluggable transports. Current version implements four
+transports: UDPv4, UDPv6, TCPv4 and TCPv6. By default, when a :class:`Participant` is created, one built-in
+UDPv4 transport is configured.
 
-You can add custom transport using the attribute ``rtps.userTransports``.
+You can add custom transports using the attribute ``rtps.userTransports``.
 
 .. code-block:: c++
```
Comment (richiware) [inline docs/pubsub.rst]: Add a table with the code and the xml configuration.

File: docs/pubsub.rst [hunk 8]
```diff
@@ -460,20 +460,136 @@ You can add custom transport using the attribute ``rtps.userTransports``.
...
+
+
+TCP Transport
+"""""""""""""
+
+To use TCP transports you need to define some more configurations:
+
+You must create a new TCP transport descriptor, for example TCPv4.
+This transport descriptor has a field named ``listening_ports`` that indicates to Fast-RTPS in which physical TCP
+port our participant will listen for input conections. If ommited, the participant will not be able to receive incoming
```
Comment (richiware) [inline docs/pubsub.rst]: Format this paragraph (dividing each line) and change `TCP ports`.

File: docs/pubsub.rst [hunk 9]
```diff
@@ -460,20 +460,136 @@ You can add custom transport using the attribute ``rtps.userTransports``.
...
+
+You can configure UDP Transports, either UDPv4 or UDPv6, from C++ or XML.
+
++---------------------------------------------------------------------------------+---------------------------------------------------------------------+
+| C++                                                                             | XML                                                                 |
++=================================================================================+=====================================================================+
+| .. code-block:: c++                                                             | .. code-block:: xml                                                 |
+|                                                                                 |                                                                     |
+|     RTPSParticipantAttributes Pparams;                                          |     <profiles>                                                      |
+|     //Create a descriptor for the new transport                                 |         <transports>                                                |
```
Comment (richiware) [inline docs/pubsub.rst]: Change `transports` to `transport_descriptors`

File: docs/pubsub.rst [hunk 10]
```diff
@@ -460,20 +460,136 @@ You can add custom transport using the attribute ``rtps.userTransports``.
...
+
+To use TCP transports you need to define some more configurations:
+
+You must create a new TCP transport descriptor, for example TCPv4.
+This transport descriptor has a field named ``listening_ports`` that indicates to Fast-RTPS in which physical TCP
+port our participant will listen for input conections. If ommited, the participant will not be able to receive incoming
+connections, but will be able to connect to others participants that have configured their listening ports.
+The transport must be added in the ``userTransports`` list of the participant attributes.
+
+To configure the participant to connect to another node through TCP, you must configure add Locator to its ``initialPeersList`` that points to the remote *listening port*.
```
Comment (richiware) [inline docs/pubsub.rst]: Divide next table in two. One configuration the participant that is the server (using the listening port), and other table with the configuration of the participant that is the client (setting the initial peers).

File: docs/pubsub.rst [hunk 11]
```diff
@@ -460,20 +460,136 @@ You can add custom transport using the attribute ``rtps.userTransports``.
...
+
+**IPLocator**
+
+IPLocator is an auxiliary static class that offers methods to ease the management of IP based locators, as UDP or TCP.
+In TCP, the port field of the locator is divided into physical and logical port.
+Physical port is the port used by the network device, the real port that the operating system understands.
+Logical port can be seen as RTPS port, or UDP's equivalent port (physical ports of UDP, are logical ports in TCP).
+Logical ports normally are not neccesary to manage explicitely, but you can do it through IPLocator class.
+Physical ports instead, must be set to explicitely use certain ports, to allow the communication through a NAT, for
+example.
```
Comment (richiware) [inline docs/pubsub.rst]: Write examples of code.

**Pull request merged**